src/reactionsystem.jl

```diff
@@ -1660,6 +1672,7 @@ function DiffEqBase.ODEProblem(rs::ReactionSystem, u0, tspan,
     pmap = symmap_to_varmap(rs, p)
     osys = convert(ODESystem, rs; name, combinatoric_ratelaws, include_zero_odes, checks,
                    remove_conserved)
+    osys = complete(osys)
```

Comment:
Error for non-compelte `ReactionSystem` input is already thrown in the convert call. This converts the osys (or similar for the other cases) to convert so that it can be simulated.

Reply:
Good point, didn't directly think of the extensions. Both BifurcationKit/HomotopyContionuation should throw an error if applied to non-complete systems. I can add a tests for this though (or add an error at a higher level, so it is thrown already in `BifurcationProblem`/`hc_steady_states`, rather than in the created `NonlinearProblem`.

I could make StructuralIdentifiability require completeness. But think it does not do that for MTK models. I will double check what SI does when you given it a normal MTK system, and we use the same for Catalyst's `ReactionSystem`s, that would make sense right?